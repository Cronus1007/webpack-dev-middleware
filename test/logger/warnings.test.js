import express from 'express';
import request from 'supertest';

import middleware from '../../src';

import getCompiler from '.././helpers/getCompiler';
import GetLogsPlugin from '.././helpers/GetLogsPlugin';

import webpackWarningConfig from '.././fixtures/webpack.warning.config';
import webpackMultiWarningConfig from '.././fixtures/webpack.array.warning.config';

describe('should logging an warning', () => {
  let compiler;
  let getLogsPlugin;
  let instance;
  let app;
  let listen;

  function listenShorthand(done) {
    return app.listen((error) => {
      if (error) {
        return done(error);
      }

      return done();
    });
  }

  function close(done) {
    if (instance.context.watching.closed) {
      if (listen) {
        listen.close(done);
      } else {
        done();
      }
    }
  }
  beforeAll((done) => {
    compiler = getCompiler(webpackWarningConfig);

    getLogsPlugin = new GetLogsPlugin();
    getLogsPlugin.apply(compiler);

    instance = middleware(compiler);

    app = express();
    app.use(instance);

    listen = listenShorthand(done);
  });

  afterAll(close);

  it('should logging', (done) => {
    request(app)
      .get('/bundle.js')
      .expect(200, (error) => {
        if (error) {
          return done(error);
        }

        instance.invalidate();

        return instance.waitUntilValid(() => {
          expect(getLogsPlugin.logs).toMatchSnapshot();

          done();
        });
      });
  });
});

describe('should logging warnings in multi-compiler mode', () => {
  let compiler;
  let getLogsPlugin;
  let instance;
  let app;
  let listen;

  function listenShorthand(done) {
    return app.listen((error) => {
      if (error) {
        return done(error);
      }

      return done();
    });
  }

  function close(done) {
    if (instance.context.watching.closed) {
      if (listen) {
        listen.close(done);
      } else {
        done();
      }
    }
  }
  beforeAll((done) => {
    compiler = getCompiler(webpackMultiWarningConfig);

    getLogsPlugin = new GetLogsPlugin();
    getLogsPlugin.apply(compiler);

    instance = middleware(compiler);

    app = express();
    app.use(instance);

    listen = listenShorthand(done);
  });

  afterAll(close);

  it('should logging', (done) => {
    request(app)
      .get('/static-one/bundle.js')
      .expect(200, (error) => {
        if (error) {
          return done(error);
        }

        instance.invalidate();

        return instance.waitUntilValid(() => {
          expect(getLogsPlugin.logs).toMatchSnapshot();

          done();
        });
      });
  });
});
