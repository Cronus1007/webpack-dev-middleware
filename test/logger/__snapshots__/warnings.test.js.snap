// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should logging an warning should logging 1`] = `
Array [
  Array [
    "webpack-dev-middleware",
    "info",
    Array [
      "wait until bundle finished: /bundle.js",
    ],
  ],
  Array [
    "webpack-dev-middleware",
    "warn",
    Array [
      "WARNING",
    ],
  ],
  Array [
    "webpack-dev-middleware",
    "info",
    Array [
      "Compiled with warnings.",
    ],
  ],
  Array [
    "webpack-dev-middleware",
    "info",
    Array [
      "Compiling...",
    ],
  ],
  Array [
    "webpack-dev-middleware",
    "info",
    Array [
      "wait until bundle finished",
    ],
  ],
  Array [
    "webpack-dev-middleware",
    "warn",
    Array [
      "WARNING",
    ],
  ],
  Array [
    "webpack-dev-middleware",
    "info",
    Array [
      "Compiled with warnings.",
    ],
  ],
]
`;

exports[`should logging warnings in multi-compiler mode should logging 1`] = `
Array [
  Array [
    "webpack-dev-middleware",
    "info",
    Array [
      "wait until bundle finished: /static-one/bundle.js",
    ],
  ],
  Array [
    "webpack-dev-middleware",
    "warn",
    Array [
      "WARNING",
    ],
  ],
  Array [
    "webpack-dev-middleware",
    "info",
    Array [
      "Compiled with warnings.",
    ],
  ],
  Array [
    "webpack-dev-middleware",
    "warn",
    Array [
      "WARNING",
    ],
  ],
  Array [
    "webpack-dev-middleware",
    "info",
    Array [
      "Compiled with warnings.",
    ],
  ],
  Array [
    "webpack-dev-middleware",
    "info",
    Array [
      "Compiling...",
    ],
  ],
  Array [
    "webpack-dev-middleware",
    "info",
    Array [
      "wait until bundle finished",
    ],
  ],
  Array [
    "webpack-dev-middleware",
    "warn",
    Array [
      "WARNING",
    ],
  ],
  Array [
    "webpack-dev-middleware",
    "info",
    Array [
      "Compiled with warnings.",
    ],
  ],
  Array [
    "webpack-dev-middleware",
    "warn",
    Array [
      "WARNING",
    ],
  ],
  Array [
    "webpack-dev-middleware",
    "info",
    Array [
      "Compiled with warnings.",
    ],
  ],
]
`;
